tests/publiccloud/run_ltp.pm

```diff
@@ -161,6 +161,19 @@ sub upload_ltp_logs
     die $@ if $@;
 }
 
+sub dump_kernel_config
+{
+    my ($self, $instance) = @_;
+
+    record_info("uname -a", $instance->run_ssh_command(cmd => "uname -a"));
+
+    my $uname_r = $instance->run_ssh_command(cmd => "uname -r");
+    chomp $uname_r;
+
+    record_info("KERNEL CONFIG", $instance->run_ssh_command(cmd => 'cat /boot/config-$uname_r'));
```

Comment:
I wonder why you did not call `uname -r` in the single ssh command (instead of 2 calls), i.e. `cat /boot/config-$(uname -r)`.
IMHO this should work:
```
$instance->run_ssh_command(cmd => 'cat /boot/config-$(uname -r)'));
```
And if not, it'd just require some escaping (`$` and or the brackets).